apps/generator/docs/generator-template.md

```diff
@@ -643,10 +643,558 @@ Temperature drop detected 66943992 sent to temperature/dropped
+
+### java - package.json file
+Add the following code snippet to your package.json file:
+
+```json
+{
+    "name": "java-mqtt-client-template",
+    "version": "0.0.1",
+    "description": "A template that generates a java MQTT client using MQTT.",
+    "generator": {
+      "renderer": "react",
+      "apiVersion": "v1",
+      "generator": ">=1.10.0 <2.0.0",
+      "supportedProtocols": ["mqtt"]
+    },
+    "dependencies": {
+      "@asyncapi/generator-react-sdk": "^0.2.25"
+    },
+    "devDependencies": {
+      "rimraf": "^5.0.0"
+    }
+}
+```
+
+Navigate to the **java-mqtt-client-template** directory. Run the command `npm install` on your terminal to install the dependencies specified in **package.json**.
+
+### java - index.js file
+
+The **index.js** file is used to define the logic for your template. Inside the template folder, create an **index.js** file and add the code snippet below:
+
+```js
+//1
+import { File } from '@asyncapi/generator-react-sdk'
+//2
+export default function ({ asyncapi }) {
+//3
+  return <File name="Client.java">{asyncapi.info().title()}</File>
+}
+```
+
+To see this in action, navigate to the **java-mqtt-client-template** directory. Then, run `asyncapi generate fromTemplate src/fixtures/asyncapi.yml ./ --output src/main/java` command on your terminal. Should get the same sucess message and a **Client.java** file in **src/main/java**.
+
+**NOTE: Nothing different has occurred so far**
+
+### 2. Create the java client
+
+  ##### a. Setting up gradle
+The first step in creating the Java client to send messages using the MQTT protocol is to ensure that your build.gradle file includes the correct dependencies Add the code snippet below into your **build.gradle** file.
+
+```groovy
+plugins {
+    id 'java'
+    id 'application'
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'org.eclipse.paho:org.eclipse.paho.client.mqttv3:1.2.5'
+}
+
+application{
+    mainClass = project.hasProperty('mainClass') ? project.mainClass : 'Client'  // Default to 'Client' if no property is passed
+}
+```
+
+Here's what is contained in the code snippet above:
+
+- **pluggins** - This section defines the plugins applied to the Gradle project.
+  - **id 'java'** - This applies the Java plugin, which allows you to compile and run Java code.
+  - **id 'application'** - This plugin is used to support building and running applications. It helps with creating an executable JAR.
+- **repositories** - This section tells you app to fetch dependencies from Maven Central to retrieve the MQTT client library.
+- **dependencies** This specifies that the project depends on the Eclipse Paho MQTT client version 1.2.5 and is needed to compile and run.
+- **application** - This section defines how the application should be executed. **mainClass** specifies the main class to be executed in a java application. It can be specificec via CL, else it defaults to **Client.java** file.
```

Comment:
```suggestion
- **application** - This section defines how the application should be executed. **mainClass** specifies the main class to be executed in a Java application. It can be specified via the command line else it defaults to the **Client.java** file.
```